modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/rel/set/IgniteMapIntersect.java

```diff
@@ -70,10 +76,15 @@ public <T> T accept(IgniteRelVisitor<T> visitor) {
         return visitor.visit(this);
     }
 
+
+
     /** {@inheritDoc} */
     @Override
     protected RelDataType deriveRowType() {
-        return buildRowType();
+        IgniteTypeFactory typeFactory = (IgniteTypeFactory) getCluster().getTypeFactory();
+        RelDataType rowType = getInput(0).getRowType();
+
+        return IgniteMapSetOp.buildRowType(typeFactory, rowType, getInputs().size());
```

Comment:
I guess `int aggregateFieldsCount() {` now can return  just `getRowType().getFieldsCount()`

https://github.com/apache/ignite-3/pull/2443/files#diff-20e473731eb1b98eda09bb76982443f7b3ae982121eac100d4b3effbad6611deR91-R94

Reply:
This line does not change this number remains the same : 
> line(set_operand_columns) + len(inputs) 

See explanation on how this operator works in `IntersectNode`.